packages/angular/src/operators/from-adapter-event.ts

```diff
@@ -0,0 +1,14 @@
+import { SignerWalletAdapter, WalletAdapter, WalletAdapterEvents } from '@solana/wallet-adapter-base';
+import { fromEventPattern, Observable } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+
+export const fromAdapterEvent =
+    (eventName: keyof WalletAdapterEvents) => (source: Observable<WalletAdapter | SignerWalletAdapter>) =>
```

Comment:
What does `SignerWalletAdapter` do here?

Reply:
This is a custom RxJs Observable that maps an adapter event listener into an observable, which is pretty handy. This operator would receive an adapter which can be `WalletAdapter | SignerWalletAdapter | MessageSignerWalletAdapter`, I believe the compiler was crying at me for some reason and I had to add this, let me double-check because it's weird that I didn't include `MessageSignerWalletAdapter` and yet it works

Reply:
Huh, okay. I would think that just `WalletAdapter` would work since the others inherit.